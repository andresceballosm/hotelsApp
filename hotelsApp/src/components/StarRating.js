import React, { Component } from 'react';
import {
	StyleSheet,
	View,
	Image
} from 'react-native';

export default class StarRating extends Component {
	render() {
        let ratings = this.props.ratings;
        let stars = [];

        for (var i = 1; i <= 5; i++) {
			let path = require('../../assets/icons/star-gold.png');
			if (i > ratings) {
				path = require('../../assets/icons/star-white.png');
			}

			stars.push((<Image key={i} style={styles.image} source={path} />));
        }
        
		return (
			<View style={ styles.container }>
              { stars }
			</View>
		);
	}
}

const styles = StyleSheet.create({
    container: {
		flexDirection: 'row',
		alignItems: 'center'
	},
	image: {
		width: 25,
		height: 25
	},
});