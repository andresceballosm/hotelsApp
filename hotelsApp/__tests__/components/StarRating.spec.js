import 'react-native';
import React from 'react'
import renderer from 'react-test-renderer';
import StarRating from '../../src/components/StarRating';

it('renders correctly', () => {
  const starRating = renderer.create(
    <StarRating/>
  ).toJSON();
  
  expect(starRating).toMatchSnapshot();
});