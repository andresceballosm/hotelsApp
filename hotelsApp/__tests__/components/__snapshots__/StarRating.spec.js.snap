// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/star-gold.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/star-gold.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/star-gold.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/star-gold.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/star-gold.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
</View>
`;
